sdk/core/Azure.Core/tests/ETagTests.cs

```diff
@@ -76,5 +77,27 @@ public void EqualityMembersDefault()
 
             Assert.True(eTag.Equals((object)new ETag(null)));
         }
+
+        [Test]
+        public void ThrowsForEtagsWithoutQuotes()
+        {
+            Assert.Throws<ArgumentException>(() => ETag.Parse("lalala"));
+        }
+
+        [Test]
+        public void ThrowsForParseWeakEtag()
+        {
+            Assert.Throws<NotSupportedException>(() => ETag.Parse("W/\"lalala\""));
+        }
+
+        [Theory]
```

Comment:
Tip: doesn't hurt, but NUnit (since some older version) doesn't require `[Theory]` if you have `[TestCase]` attributed.